src/entities/base.js

```diff
@@ -0,0 +1,19 @@
+class Base {
+    constructor(attr) {
+        this.attr = attr
+    }
+
+    get id() {
+        return this.attr.id
+    }
+
+    toString() {
+        return `${this.constructor.name}#${this.id}`
+    }
+
+    toJSON() {
+        return this.attr
```

Comment:
Do I understand correctly that methods in the Base class are "default" ones?
So every instance of let's say class Location has the "toJSON" method, even though it's not defined in the Location class itself?

Reply:
Exactly